tools/readme_config_includer/generator.go

```diff
@@ -257,6 +257,6 @@ func main() {
 	}
 	defer file.Close()
 	if _, err := output.WriteTo(file); err != nil {
-		log.Fatalf("Writing output file failed: %v", err)
+		log.Panicf("Writing output file failed: %v", err)
```

Comment:
Which rule required this change? Don't these both fatal and panic both stop operation when called?

Reply:
@powersj 
```
tools/readme_config_includer/generator.go:260:3  gocritic  exitAfterDefer: log.Fatalf will exit, and `defer file.Close()` will not run
```

They both do, but:

"`Fatalf` is equivalent to `Printf()` followed by a call to `os.Exit(1)`." (https://pkg.go.dev/log#Fatalf)
"`Exit` causes the current program to exit with the given status code. Conventionally, code zero indicates success, non-zero an error. The program terminates immediately; **deferred functions are not run**." (https://pkg.go.dev/os#Exit)

vs.

"`Panicf` is equivalent to `Printf()` followed by a call to `panic()`." (https://pkg.go.dev/log#Panicf)
"`Panic` is a built-in function that stops the ordinary flow of control and begins panicking. When the function `F` calls `panic`, execution of `F` stops, **any deferred functions in `F` are executed normally**, and then `F` returns to its caller. To the caller, `F` then behaves like a call to `panic`. The process continues up the stack until all functions in the current goroutine have returned, at which point the program crashes." (https://go.dev/blog/defer-panic-and-recover)